arch/x86_64/fentry.S

```diff
@@ -1,8 +1,55 @@
 /* argument passing: %rdi, %rsi, %rdx, %rcx, %r8, %r9 */
 /* return value: %rax */
 /* callee saved: %rbx, %rbp, %rsp, %r12-r15 */
-/* stack frame (with -pg): return addr = 8(%rbp), prev fp = 0(%rbp) */
-/* stack frame (with -fentry): return addr = (%rsp), prev fp = 8(%rsp) */
+/* stack frame (with -pg -mfentry): parent addr = 56(%rsp), child addr = 48(%rsp) */
```

Comment:
1. I meant "child addr == child ip".

2. Really?  %rsp is a stack **pointer** so its value should point to a stack address.  Dereferencing %rsp can result in a return address (of a function).

For example:
* %rsp = 0x7fffff8720
* (%rsp) = 0x4006ab